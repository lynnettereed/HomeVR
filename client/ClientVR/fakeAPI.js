const fakeAPI = {
  domMenuContent: {
    menuFamilyRoom: {
      menuData: [
        {
          sectionHeader: 'scene',
          sectionOptions: ['FamilyRoom', 'Kitchen'],
        },
        {
          sectionHeader: 'elevation',
          sectionOptions: ['american classic', 'bella vista', 'bella vista brick'],
        },
        {
          sectionHeader: 'sunroom',
          sectionOptions: ['on', 'off'],
        },
      ],
    },
    menuKitchen: {
      menuData: [
        {
          sectionHeader: 'scene',
          sectionOptions: ['FamilyRoom', 'Kitchen'],
        },
        {
          sectionHeader: 'elevation',
          sectionOptions: ['american classic', 'bella vista', 'bella vista brick'],
        },
        {
          sectionHeader: 'sunroom',
          sectionOptions: ['on', 'off'],
        },
        {
          sectionHeader: 'cabinets',
          sectionOptions: ['option1', 'option2', 'option3', 'option4'],
        },
        {
          sectionHeader: 'backsplash',
          sectionOptions: ['option1', 'option2', 'option3', 'option4'],
        },
        {
          sectionHeader: 'counter',
          sectionOptions: ['option1', 'option2', 'option3', 'option4'],
        },
      ],
    }
  },
  modalContent: {
    appliances: {
      refrigerator: {
        data: {
          title: '36-inch Wide Side-by-Side Refrigerator - 28 cu. ft.',
          model: 'WRS588FIHV',
          msrp: '$1,899.00',
          quickFeatures: [
            '-Exterior Ice with EveryDrop™ Filtration',
            '-In-Door-Ice® Storage',
            '-Accu-Chill™ Temperature Management System'
          ],
          additionalFeatures: [
            {
              title: 'Exterior Ice and Water Dispenser with EveryDrop™ Filtration',
              info: 'Access fresh filtered water and ice without ever opening the refrigerator door.'
            },
            {
              title: 'In-Door-Ice® Storage',
              info: 'Get an extra full shelf in the freezer with an ice bin that\'s been moved to the door.'
            },
            {
              title: 'Accu-Chill™ Temperature Management System',
              info: 'Cool food quickly with technology that senses and adapts to create the ideal environment for food.'
            },
            {
              title: 'Frameless Glass Shelves',
              info: 'Store more items on each shelf with wall-to-wall frameless glass shelves, which offer greater storage flexibility.'
            }
          ],
          imgSrc: '/static_assets/images/refrigerator.png',
          prodImgSrc: './static_assets/images/refrigerator.png',
          colorOptions: [
            {
              name: 'black',
              value: '#000'
            },
            {
              name: 'fingerprint resistant black stainless',
              value: '#5B5959'
            },
            {
              name: 'white',
              value: '#FFF'
            },
            {
              name: 'fingerprint resistant stainless steel',
              value: '#D8D8D8'
            },
          ]
        }
      }
    }
  },
  getMenuData: function() {
    return new Promise((resolve, reject) => {
      const menuDescriptor = Object.getOwnPropertyDescriptor(this, 'domMenuContent');
      menuDescriptor ? resolve(menuDescriptor.value) : reject(new Error('menu content not found'));
    });
  },
  getModalData: function() {
    return new Promise((resolve, reject) => {
      const modalDescriptor = Object.getOwnPropertyDescriptor(this, 'modalContent');
      modalDescriptor ? resolve(modalDescriptor.value) : reject(new Error('modal content not found'));
    });
  }
}

export default fakeAPI;
