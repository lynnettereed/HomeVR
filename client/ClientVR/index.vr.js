import React from 'react';
import {
  AppRegistry,
  asset,
  Pano,
  View,
  NativeModules,
  StyleSheet,
  Text,
  VrButton,
  VrHeadModel
} from 'react-vr';

import TextboxVr from './components/TextboxVr';

const domTextboxContent = {
  header: 'This is a DOM Overlay!',
  description: 'A dom overlay is a 2D window that takes over the 3D world, allowing for better interactivity and content consumption outside of VR. DOM Overlays are implemented as Native Modules, for more info on native modules, check the following links:',
}

const vrTextboxContent =
  'This is a React VR textbox! This is how you would show text in VR, where DOM Overlay is not accessible.';

export default class ClientVR extends React.Component {
  constructor() {
    super();

    this.state = {
      renderVrTextbox: false,
    };

    this._toggleDisplay = this.toggleDisplay.bind(this);
  }

  // Determine whether content should be displayed on the dom overlay, or as a
  // react-vr component based on VrHeadModel's inVR API.
  toggleDisplay() {
    if (VrHeadModel.inVR()) {
      this.setState({renderVrTextbox: !this.state.renderVrTextbox});
    } else {
      // Not in VR, use the dom overlay
      NativeModules.DomOverlayModule.openOverlay(domTextboxContent);
    }
  }

  render() {
    return (
      <View style={ styles.rootView }>
        <Pano source={ asset('chess-world.jpg') } />
        <View style={ styles.triggerContainer }>
          <VrButton style={ styles.triggerButton } onClick={ this._toggleDisplay }>
            <Text style={ styles.triggerText }>Click Me!</Text>
          </VrButton>
        </View>
        {this.state.renderVrTextbox && <TextboxVr text={ vrTextboxContent } />}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  rootView: {
    layoutOrigin: [0.5, 0.5],
    position: 'absolute',
  },
  triggerContainer: {
    transform: [{rotateY: 0}, {translateZ: -3}],
  },
  triggerButton: {
    borderRadius: 0.05,
    height: 1,
    width: 1,
    backgroundColor: '#F00',
    justifyContent: 'center',
  },
  triggerText: {
    alignSelf: 'center',
    fontSize: 0.2,
    textAlign: 'center',
    textAlignVertical: 'center',
  },
});

AppRegistry.registerComponent('ClientVR', () => ClientVR);
