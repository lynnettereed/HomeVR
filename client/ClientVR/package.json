{
  "name": "ClientVR",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node -e \"console.log('open browser at http://localhost:8081/vr/\\n\\n');\" && node node_modules/react-native/local-cli/cli.js start",
    "bundle": "node node_modules/react-vr/scripts/bundle.js",
    "open": "node -e \"require('xopen')('http://localhost:8081/vr/')\"",
    "devtools": "react-devtools",
    "test": "jest"
  },
  "dependencies": {
    "ovrui": "~2.0.0",
    "react": "16.0.0",
    "react-dom": "^16.2.0",
    "react-native": "~0.48.0",
    "react-native-axios": "^0.17.1",
    "react-vr": "~2.0.0",
    "react-vr-web": "~2.0.0",
    "three": "^0.87.0"
  },
  "devDependencies": {
    "babel-jest": "^19.0.0",
    "babel-preset-react-native": "^1.9.1",
    "enzyme": "^3.3.0",
    "enzyme-adapter-react-16": "^1.1.1",
    "enzyme-to-json": "^3.3.1",
    "jest": "^19.0.2",
    "react-devtools": "^2.5.2",
    "react-test-renderer": "16.0.0",
    "sinon": "^4.2.1",
    "xopen": "1.0.0"
  },
  "jest": {
    "preset": "react-vr",
    "snapshotSerializers": [
      "./node_modules/enzyme-to-json/serializer"
    ]
  }
}
