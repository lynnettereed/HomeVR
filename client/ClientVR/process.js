/**
 * The examples provided by Oculus are for non-commercial testing and
 * evaluation purposes only.
 *
 * Oculus reserves all rights not expressly granted.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
 * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON INFRINGEMENT. IN NO EVENT SHALL
 * OCULUS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
 * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
 * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 */

// Hack for react-dom to work, otherwise the console spits out
// a `process is not defined error`, in what looks like react's initialization:
// if (process.env.NODE_ENV === 'production') {
//   module.exports = require('./cjs/react.production.min.js');
// } else {
//   module.exports = require('./cjs/react.development.js');
// }
window.process = {
  env: {
    NODE_ENV: 'production',
  },
};

if (__DEV__) {
  window.process.env.NODE_ENV = 'development';
}
