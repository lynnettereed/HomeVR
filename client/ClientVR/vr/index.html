<html>
  <head>
    <title>ClientVR</title>
    <meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no">
    <!-- Font Awesome css -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" type="text/css" />
    <!-- Bootstrap css -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta.3/css/bootstrap.min.css" type="text/css">
    <!-- Custom css -->
    <link rel="stylesheet" href="../static_assets/css/style.css" type="text/css" />
    <style>
      body {
        margin: 0;
      }
    </style>
  </head>
  <body>
    <section class="section-main">

      <div id="outer-menu">
        <form>
          <fieldset class="form-group">
            <legend>Elevation</legend>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="elevation" value="AmericanClassic" checked="checked">
              <label class="form-check-label">American Classic</label>
            </div>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="elevation" value="BellaVista">
              <label class="form-check-label">Bella Vista</label>
            </div>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="elevation" value="BellaVistaBrick">
              <label class="form-check-label">Bella Vista Brick</label>
            </div>
          </fieldset>

          <fieldset class="form-group">
            <legend>Garage</legend>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="garage" value="2Car" checked="checked">
              <label class="form-check-label">2 Car</label>
            </div>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="garage" value="3Car">
              <label class="form-check-label">3 Car</label>
            </div>
          </fieldset>

          <fieldset class="form-group">
            <legend>Sunroom</legend>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="sunroom" value="On" checked="checked">
              <label class="form-check-label">On</label>
            </div>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="sunroom" value="Off">
              <label class="form-check-label">Off</label>
            </div>
          </fieldset>

          <fieldset class="form-group">
            <legend>View</legend>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="view" value="FrontRight" checked="checked">
              <label class="form-check-label">Front Right</label>
            </div>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="view" value="FrontLeft">
              <label class="form-check-label">Front Left</label>
            </div>
            <div class="form-check">
              <input class="form-check-input" type="radio" name="view" value="Back">
              <label class="form-check-label">Back</label>
            </div>
          </fieldset>
        </form>
        <button id="vr-btn">Init VR</button>
      </div>

      <div id="media-container">
        <div class="img-container">
          <img id="baseline" src="../static_assets/images-2D/Foster_Ext_FrontRight_AmericanClassic.jpg" alt="">
        </div>
      </div>
    </section>
    <script
      src="https://code.jquery.com/jquery-3.2.1.min.js"
      integrity="sha256-hwg4gsxgFZhOsEEamdOYGBf13FyQuiTwlAQgxVSNgt4="
      crossorigin="anonymous"></script>
    <!-- When you're ready to deploy your app, update this line to point to your compiled client.bundle.js -->
    <script src="./client.bundle?platform=vr"></script>
    <script>
      // Initialize the React VR application
      // ReactVR.init(
      //   //When you're ready to deploy your app, update this line to point to
      //   //your compiled index.bundle.js
      //   '../index.vr.bundle?platform=vr&dev=true',
      //   //Attach it to the body tag
      //   //document.body
      //   document.getElementById('vr-container')
      // );
    </script>
  </body>
</html>
