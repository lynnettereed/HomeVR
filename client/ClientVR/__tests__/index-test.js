import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import 'react-native';
import 'react-vr';
import { NativeModules } from 'react-native';
import { shallow } from 'enzyme';
import React from 'react';
import ClientVR from '../index.vr.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('NativeModules', () => {
  return {
    DomOverlayModule: {
      openPersistent: jest.fn(),
    },
  };
});

describe('<ClientVR />', () => {

  it('renders correctly', () => {
    const wrapper = shallow(
      <ClientVR />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
