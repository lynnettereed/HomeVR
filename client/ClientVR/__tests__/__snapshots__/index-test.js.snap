// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ClientVR /> renders correctly 1`] = `
<View>
  <FamilyRoom
    elevation="american classic"
    renderVrMenu={false}
    sunroomOn={false}
  />
</View>
`;
